rust/numaflow-monitor/src/runtime.rs

```diff
@@ -0,0 +1,339 @@
+            match fs::read_dir(&sub_dir_path) {
+                Err(e) => {
+                    error!(
+                        "{}",
+                        Error::File(format!("Failed to read subdirectory: {:?}", e))
+                    );
+                    continue;
+                }
+                Ok(file_paths) => {
+                    for file_entry in file_paths.flatten() {
+                        // process content of each file into error entry
+                        if let Err(e) = process_file_entry(&file_entry, &mut errors) {
+                            error!(
+                                "{}",
+                                Error::File(format!(
+                                    "error: {} in processing file entry: {:?}",
+                                    e,
+                                    file_entry.file_name()
+                                ))
+                            );
+                            continue;
+                        }
+                    }
+                }
+            }
+        }
+        Ok(errors)
+    }
+}
+
+/// extracts the container information from the error message
+fn extract_container_name(error_message: &str) -> String {
+    let re = Regex::new(r"\((.*?)\)").unwrap();
+    re.captures(error_message)
+        .and_then(|caps| caps.get(1).map(|m| m.as_str().to_string()))
+        .expect(
+            format!(
+                "Failed to extract container name from the error message: {}",
+                error_message
+            )
+            .as_str(),
+        )
+}
+
+/// Converts gRPC status to a JSON object.
+fn grpc_status_to_json(grpc_status: &Status, container_name: &str, timestamp: i64) -> String {
+    // Extract code, message, and details from gRPC Status
+    let code = grpc_status.code().to_string();
+    let message = grpc_status.message().to_string();
+    let details_bytes = grpc_status.details();
+
+    // Convert status details from bytes to a string
+    let details_str = String::from_utf8_lossy(details_bytes);
+
+    // Convert timestamp to RFC 3339 string
+    let datetime = DateTime::<Utc>::from_timestamp(timestamp, 0).unwrap();
+    let rfc3339_timestamp = datetime.to_rfc3339();
+    // Create a RuntimeErrorEntry instance
+    let runtime_error_entry = RuntimeErrorEntry {
+        container: container_name.to_string(),
+        timestamp: rfc3339_timestamp,
+        code,
+        message,
+        details: details_str.to_string(),
+    };
+
+    // Serialize the RuntimeErrorEntry instance to a JSON string
+    serde_json::to_string(&runtime_error_entry).expect("Failed to serialize runtime error message")
+}
+
+fn process_file_entry(
+    file_entry: &fs::DirEntry,
+    errors: &mut Vec<RuntimeErrorEntry>,
+) -> Result<()> {
+    let file_path = file_entry.path();
+    let file_name_to_ignore = Some(std::ffi::OsStr::new("current.json"));
+    // if file path isn't a file or file name is current.json, continue processing other files
```

Comment:
implement TryFrom on runtime error entry? to create runtime entry type from bytes